src/snarky.d.ts

```diff
@@ -460,18 +460,19 @@ declare class Bool {
   static count(x: Bool | boolean[]): Field;
 
   static sizeInFields(): number;
-  static toFields(x: Bool | boolean): Field[];
+  static toFields(x: Bool): Field[];
   static ofFields(fields: Field[]): Bool;
 
   static toJSON(x: Bool): JSONValue;
   static fromJSON(x: JSONValue): Bool | null;
+  static check(x: Bool): void;
 }
 
 declare interface AsFieldElements<T> {
   toFields(x: T): Field[];
   ofFields(x: Field[]): T;
   sizeInFields(): number;
-  check?: (x: T) => void;
+  check(x: T): void;
```

Comment:
I made this change to facilitate inferring `<T>` from and `AsFieldElements<T>`.
That every circuit type has a `check` method is now also enforced by `Circuit.witness`.
@imeckler had mentioned that he should've made `check` required instead of optional, so I went with that; it makes for simpler types and less special-casing.
This change has minor repercussions throughout the code base

Reply:
Yeah you're right. It was made to enable some typing that I intended to add in this PR, but ended up saving for the next PR.